src/middleware/user.ts

```diff
@@ -0,0 +1,38 @@
+import { defineMiddleware } from 'astro:middleware';
+import { url, api } from '@lib/helpers';
+
+export const user = defineMiddleware(async ({ locals, request, redirect }, next) => {
+	// If page url starts with '/u', then we know this is a user page.
+	if (!new URL(request.url).pathname.startsWith('/u')) {
+		return next();
+	}
+
+	// Check if the user is authenticated.
+	if (!locals.isAuthenticated) {
+		// If not, redirect to the login page.
+		return redirect(url('auth/login'));
+	}
+
+	// Get User data from the API.
+	const authenticationHeader = locals.authHeader as string;
+	const userResponse = await (
+		await fetch(api('auth/user'), {
+			method: 'GET',
+			headers: {
+				Authorization: authenticationHeader,
+			},
+		})
+	).json();
```

Comment:
Why not just decode accesstoken?

Reply:
**We need name and email and we only save userId in the tokens.**

But it's not a bad idea to add name and email. That would make it so we wouldn't have to contact the backend at load. I'll make issues for that.